plugins/plugin-chart-echarts/src/Tree/transformProps.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ChartProps, getMetricLabel } from '@superset-ui/core';
+import { EChartsOption, TreeSeriesOption } from 'echarts';
+import { TreeSeriesNodeItemOption } from 'echarts/types/src/chart/tree/TreeSeries';
+import {
+  EchartsTreeFormData,
+  DEFAULT_FORM_DATA as DEFAULT_GRAPH_FORM_DATA,
+  TreeDataRecord,
+} from './types';
+import { DEFAULT_TREE_SERIES_OPTION } from './constants';
+import { EchartsProps } from '../types';
+import { OptionDataValue, OptionName } from 'echarts/types/src/util/types';
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queriesData } = chartProps;
+  const data: TreeDataRecord[] = queriesData[0].data || [];
+
+  const {
+    id,
+    relation,
+    name,
+    rootNode,
+    metric = '',
+    layout,
+    orient,
+    symbol,
+    symbolSize,
+    roam,
+    position,
+    emphasis,
+  }: EchartsTreeFormData = { ...DEFAULT_GRAPH_FORM_DATA, ...formData };
+
+  const metricLabel = getMetricLabel(metric);
+  const tree: TreeSeriesNodeItemOption = { name: rootNode, children: [] };
+  const indexMap: { [name: string]: number } = {};
+  let rootNodeId: null | string = null;
+  let nameColumn: string;
+  if (name) {
+    nameColumn = name;
+  } else {
+    nameColumn = id;
+  }
+  for (let i = 0; i < data.length; i++) {
+    const nodeId = data[i][id] as string;
+    indexMap[nodeId] = i;
+    data[i].children = [] as TreeSeriesNodeItemOption[];
+    if (data[i][nameColumn] == rootNode) {
+      rootNodeId = nodeId;
+    }
+  }
+
+  if (rootNodeId) {
+    data.forEach(node => {
+      if (node[relation] === rootNodeId) {
+        tree.children!.push({
+          name: node[nameColumn] as OptionName,
+          children: node.children as TreeSeriesNodeItemOption[],
+          value: node[metricLabel] as OptionDataValue,
```

Comment:
@villebro  Not able to make ts compiler take type from data.

Reply:
I've pushed a commit that addresses this and removes all the `as` clauses from this file. I'll open it up as a PR into your branch, and you can review and merge it if it looks good 😁

Reply:
I have a separate concern with this code - what happens if the user chooses "children" as the name column or the metric label? It seems like this code would break. Is this scenario actually possible, and if so what can we do to guard against it?

Reply:
@mayurnewase could you review and merge that PR if it looks good? Thanks!

Reply:
@rusackas yeah I updated those changes manually because I had to change it little bit.